core/src/main/java/com/linecorp/armeria/client/http/DefaultHttpClient.java

```diff
@@ -46,7 +46,10 @@ public HttpResponse execute(HttpRequest req) {
     }
 
     private HttpResponse execute(EventLoop eventLoop, HttpRequest req) {
-        return execute(eventLoop, req.method().name(), req.path(), "", req, cause -> {
+        final String respectingClientUriPath = uri().isOpaque() ? req.path() : uri().getPath() + req.path();
+        req.path(respectingClientUriPath);
```

Comment:
- Could we avoid resulting in the path that contains `//`? i.e. check `uri().getPath()` does not end with `/` and `req.path()` starts with `/`, etc
- `respectingClientUriPath` -> How about just `path`?